common/dev/snoop.c

```diff
@@ -120,10 +110,21 @@ void snoop_read()
 		LOG_ERR("snoop read buffer alloc fail");
 		return;
 	}
-	rc = snoop_aspeed_register_rx_callback(snoop_dev, snoop_rx_callback);
-	if (rc != 0) {
-		printk("Cannot register RX callback\n");
-		return;
+
+	while (1) {
+		rc = snoop_aspeed_read(snoop_dev, 0, snoop_data, true);
+		if (rc == 0) {
+			proc_postcode_ok = true;
+			if (!k_mutex_lock(&snoop_mutex, K_MSEC(1000))) {
+				snoop_read_buffer[snoop_read_num % SNOOP_MAX_LEN] = *snoop_data;
+				snoop_read_num++;
+				if (k_mutex_unlock(&snoop_mutex)) {
+					LOG_ERR("snoop read unlock fail");
```

Comment:
Not sure how defensive this code would be because if mutex is non-recursive its going to deadlock in case of next while invocation.

Reply:
Merging this PR for now.